### ISSUE TITLE
perf(model): Qwen3.8-Metal GDN sequence path refuses every restored prefix, pinning decode at ~1.8 tok/s

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#13179]
# perf(model): Qwen3.8-Metal GDN sequence path refuses every restored prefix, pinning decode at ~1.8 tok/s

```routing
lane: metalgemm
paths: ["internal/agent/inkernel_decode.go", "internal/agent/inkernel_planner.go", "internal/model/qwen35_prefill_q4k.go"]
expected_steps: 6
```

## Working spine
The live `fak up` Metal Qwen3.8-27B server logs `reused=0tok` on every turn because the resident `metal/qwen35-gdn-preprojected-sequence-v1` path hard-refuses any non-zero `matched` prefix and the Metal CPU-session seam has `p.backend == nil`, so the snapshot-admission branch that would populate the tree never runs.

## Current state
On the deployed launchd `com.fak.up` server (`fak-native up --headless --model 27B --context 20480`, endpoint `http://127.0.0.1:8080/v1`, model `qwen3.8-27b-q4_k_m`), every logged turn shows `reused=0tok` with `cacheable` limited to a 3-token fragment. Streaming decode is ~1.2-1.8 tok/s (coordinator witness 196 words/115.02s, TTFT 6.28s; this audit reproduced 5 tokens at TTFT 8.39s / 1.22 tok/s post-TTFT). `.fak/qwen-overnight/STATE.md` names `reused=0tok` as the blocker against goals prefill>40 / decode>7.5. Prefix-KV reuse is the decoded-throughput lever, and it is dead on the local Metal path.

## Why this is next
Every agentic turn on the local endpoint re-prefills the whole prompt, so TTFT dominates and per-turn cost is linear in transcript length. This is the single root cause behind the live token-serving gap; fixing it moves decode and TTFT together.

## Parent context
Parent: #12459 (30 tok/s epic). Related: #12341 (hybrid-state restore on persistent replacement), #11585 (live Metal low-activity), #13023 (mac-native prefix-reuse head-to-head), #13076 (measured server-side reuse).

## Core through-line
Admitted prefix state -> the resident GDN sequence forward restores it instead of refusing -> live `reused>0tok` and a measurable decode lift -> witnessed by the live `inkernel_chat` reuse line plus a regression test over the sequence path.

## Gold-plating boundary
Out of scope: new kernels, a new cache tier, or non-Metal backends. Do NOT relax the recurrent-state correctness arm merely to admit a prefix: a restore that changes logits is a regression, not a win.

## Done condition / witness
The decode-throughput blocker is closed and the restart-gated live server shows non-zero `reused` on a repeated-prefix turn.
Witness: `go test -count=1 ./internal/agent/... -run TestInKernel.*Prefix` and a live `grep 'inkernel_chat' /tmp/fak-up.err | tail -1` showing `reused>0tok`.

## Definition of done
- [ ] `inkernel_decode.go:288-302` no longer hard-fails a non-zero `matched` when it can restore the resident sequence state, OR the planner selects a path that admits prefix state for Metal Q4K hybrid.
- [ ] The `p.backend != nil` gate that skips snapshot admission on the Metal CPU-session seam is replaced by a path-specific capability check.
- [ ] A repeated-prefix turn on the live server logs `reused>0tok` and a decode rate above the pre-fix baseline.
- [ ] A regression test pins restored-prefix logits equal to a fresh full-prompt forward.

## Acceptance gate
`go test -count=1 ./internal/agent/... ./internal/model/...` passes, and the live reuse line is non-zero.

## Closure binding
Resolving commit cites this issue and carries the `(fak <leaf>)` trailer.

## Lane
metalgemm

## Likely files
- internal/agent/inkernel_decode.go
- internal/agent/inkernel_planner.go
- internal/agent/inkernel_planner_config.go
- internal/model/qwen35_prefill_q4k.go
- internal/model/prefix_snapshot_codec.go

## Expected steps
6

- Centrality: Core
- P1 Context: advanced - the live local serving blocker
- P2 Net value: advanced - non-zero prefix reuse lifts decode and TTFT
- P3 Adaptation: preserved - non-hybrid and non-Metal paths unchanged
- P4 Operations: advanced - the existing `reused=` log line is the telemetry

## Work estimate
Estimate: 8 points

## Overall completion contribution
Contribution: 8/8 points

## Completion standard
integrated

**Comment by anthony-chaudhary** [comment]
Fixed on `main` in `19002b614` (rebased from `333d27670`).

Root cause: `internal/radixkv/radixkv.go:624` installs a split-boundary KV only when `CanEvict()==nil`; the GDN hybrid returns `*RecurrentEvictUnsupportedError`, so a mid-edge split carried no restorable state — the lookup saw the prefix (`cacheable>0`) but realized `matched=0`, hence `reused=0tok` on every turn.

Fix: host-hybrid lookup falls back to `LookupSnapshotTieredContext` when the KV-clone lookup has no restorable match; `Backend==nil` snapshot restore builds a plain host session and installs `Cache.Clone()` (includes linear recurrent state); adaptive 64-token checkpoint prefill+admit.

Verification witness:
- `go test ./internal/agent/ -run TestInKernelHostHybrid -count=1` -> PASS; 192-token shared prefix reuses 192 (was 0); tenant-B reuses 0 (scoping intact); granularity 63->0 / 64->64 / 128->128.
- Negative control (revert source, keep test) reproduces `reused=0` -> the fix is real, not test-shaped.
- `go test ./internal/agent/ -count=1` ok except two pre-existing trunk failures (`TestInKernel{Exact,SameTenant}DeviceL1...`), confirmed failing identically on clean `origin/main`.
- Boundary: `go run ./cmd/fak-boundary check --all` -> PASS, 0 violations.

Caveat (tracked): the witness is synthetic-config, not physical Mac Metal; live re-witness after a rebuild+restart is the next rung. Ref #13180 (phase-claim guard), fak-private#1847 (opencode local default).
